test/e2e-secondary-network/secondary_network_test.go

```
@@ -219,6 +237,204 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+			if ok && len(ipPools) > 0 {
+				if poolName, ok := ipPools[0].(string); ok {
+					return poolName, nil
+				}
+			}
+		}
+	}
+
+	return "", fmt.Errorf("no IPPool found in annotation for pod %s", podName)
+}
+
+// getPodSecondaryIP fetches the secondary interface IP from the pod's annotation.
+func (data *testData) getPodSecondaryIP(namespace, podName string) (string, error) {
+	cmd := exec.Command("kubectl", "get", "pod", podName, "-n", namespace, "-o", "jsonpath={.metadata.annotations.k8s\\.v1\\.cni\\.cncf\\.io/networks}")
+	output, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to get pod annotations: %v", err)
+	}
+
+	var networkStatus []NetworkStatus
+	if err := json.Unmarshal(output, &networkStatus); err != nil {
+		return "", fmt.Errorf("failed to parse network status: %v", err)
+	}
+
+	// Extract the secondary network IP
+	for _, net := range networkStatus {
+		for _, iface := range net.Interfaces {
+			if iface.Name != "eth0" { // secondary interfaces are not eth0
+				return iface.IPs[0], nil // Return the first IP found
+			}
+		}
+	}
+
+	return "", fmt.Errorf("no secondary IP found for pod %s", podName)
+}
+
+// testPauseAgentDeletePodRestartAgent verifies OVS cleanup and IP release.
+func (data *testData) testPauseAgentDeletePodRestartAgent(vlanPod *testPodInfo, waitTime time.Duration) error {
+	logs.Infof("Fetching OVS ports for pod %s before deletion", vlanPod.podName)
+	beforePorts, err := data.getOVSPorts(vlanPod)
+	if err != nil {
+		return fmt.Errorf("failed to get OVS ports before deletion: %v", err)
+	}
+	logs.Infof("Fetching OVS ports for pod %s before deletion", beforePorts)
+
+	ipPoolNmae, err := data.getIPPoolFromPod(vlanPod.podName, data.e2eTestData.GetTestNamespace())
+	if err != nil {
+		fmt.Println("Failed to get ippool:", err)
+	} else {
+		fmt.Println("IPPool name:", ipPoolNmae)
+	}
+
+	vlanIP, err := data.getPodSecondaryIP("default", vlanPod.podName)
+	if err != nil {
+		fmt.Println("Error:", err)
```

Should we return the error?